### PR TITLE
Unregister engine on uninstall

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
Add logic to WiX script that calls
```
scripts/o3de.bat register --remove --this-engine
```
on uninstall and fixed two issues with unregistering the engine found along the way
1. engine was not removed from engine_paths because the engine root was passed in and validation expects `engine.json`
2. the `--this-engine` command does not pass the `args.remove` option for removal and I wanted to avoid any potential problems by passing in a path from WiX

*Testing*
- Tested the script command first to verify the register and unregister was working
- Created an installer and verified after installing it added the engine to o3de_manifest.json and after uninstalling the engine was removed from o3de_manifest.json
- Verified the script was run in the installer logs
```
MSI (s) (FC:1C) [10:29:18:310]: Executing op: ActionStart(Name=UnRegisterEngine,,)
MSI (s) (FC:1C) [10:29:18:310]: Executing op: CustomActionSchedule(Action=UnRegisterEngine,ActionType=3106,Source=C:\O3DE\0.0.0.0\,Target="C:\Windows\SysWOW64\cmd.exe" /C @"C:\O3DE\0.0.0.0\scripts\o3de.bat" register --remove --this-engine,)
```

Signed-off-by: Alex Peterson <26804013+AMZN-alexpete@users.noreply.github.com>

### CODE REVIEW

File: cmake/Platform/Windows/Packaging/PostInstallSetup.wxs [hunk 1]
```diff
@@ -61,5 +62,11 @@
             Execute="deferred"
             Impersonate="no"/>
 
+        <CustomAction Id="UnRegisterEngine"
+            ExeCommand="&quot;[SystemFolder]cmd.exe&quot; /C @&quot;[root.scripts]o3de.bat&quot; register --remove --this-engine"
```
Comment (AMZN-alexpete) [inline cmake/Platform/Windows/Packaging/PostInstallSetup.wxs:66]: This '@' character is important if we ever decide to include any quoted path arguments in the future.  Without it you get some fun quote stripping behavior

**Review by cgalvan — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

File: cmake/Platform/Windows/Packaging/PostInstallSetup.wxs (hunk 1 above)
Comment (AMZN-ScottR) [inline cmake/Platform/Windows/Packaging/PostInstallSetup.wxs:66]: Just want to confirm this won't return non-zero if the engine isn't registered, right?

Comment (AMZN-alexpete) [reply cmake/Platform/Windows/Packaging/PostInstallSetup.wxs:66]: I'll go ahead and ignore the return, that'd probably be best so it doesn't fail the uninstall process.

Comment (AMZN-ScottR) [reply cmake/Platform/Windows/Packaging/PostInstallSetup.wxs:66]: Cool, thanks for confirming. Agreed that adding the ignore would be the best safeguard against unrelated command failures.

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**